Update Renderer documentation to clarify tolerance option Canvas only

Hippl-Eric opened the pull request:
Fixes #7494 

- Renderer documentation does not clarify the [tolerance ](https://leafletjs.com/reference-1.7.1.html#renderer-tolerance ) option is only supported by Canvas (not SVG) renderer.
- Update documentation to reflect this.
- Refer to similar instance in documentation: Path [className](https://leafletjs.com/reference-1.7.1.html#path-classname) option

Close #7494.

johnd0e commented:
Our documentation is autogenerated from source comments, so the fix should be in `Renderer.js`.

> * Refer to similar instance in documentation: Path [className](https://leafletjs.com/reference-1.7.1.html#path-classname) option

Not exactly a good example for this case.

I have in mind 2 alternative ways:
1. Note that `tolerance` is defined not in `Canvas.js`. So simplest fix would be just move it there.
2. Or we can consider this as generic renderer option, which is subject to be (or not) implemented in respective renderers.
   This case requires more general wording, saying that the option requires support from renderer, and SVG lacks one.

In fact I'm not sure what way should be chosen.
May be let's start from 1 and wait for other opinions.

Hippl-Eric commented:
> Our documentation is autogenerated from source comments, so the fix should be in `Renderer.js`.

I reverted commit to incorrect file `reference-1.7.1.html`.  Awaiting direction before modifying `Renderer.js`.

> In fact I'm not sure what way should be chosen.
> May be let's start from 1 and wait for other opinions.

I can work through implementing 1, but can you please clarify what you are expecting?  As you point out, `tolerance` is not defined in `Canvas.js` - it is defined in `Renderer.js`:
https://github.com/Leaflet/Leaflet/blob/0f904a515879fcd08f69b7f51799ee7f18f23fd8/src/layer/vector/Renderer.js#L39-L41
`Canvas` does not have an `options` property (only inherited).  Are you suggesting to add an `options` property to `Canvas` with one nested property `tolerance : 0`?  Then removing the `tolerance` from `Renderer`?

johnd0e commented:
> Canvas does not have an options property (only inherited). Are you suggesting to add an options property to Canvas with one nested property tolerance : 0? Then removing the tolerance from Renderer?

That is exactly what I mean, right.

Hippl-Eric commented:
@johnd0e I completed the following changes:

1. Moved tolerance option from `Render` to `Canvas`
2. Updated `Path_clickTolerance` to handle possible renderers that do include a tolerance option (all `SVG` renderers now)

johnd0e commented on src/layer/vector/Path.js:
```diff
@@ -142,6 +142,7 @@ export var Path = Layer.extend({
 		// used when doing hit detection for Canvas layers
-		return (this.options.stroke ? this.options.weight / 2 : 0) + this._renderer.options.tolerance;
+		return (this.options.stroke ? this.options.weight / 2 : 0) +
+		(this._renderer.options.tolerance ? this._renderer.options.tolerance : 0);
```
```suggestion
		  (this._renderer.options.tolerance || 0);
```
